File: 백준/Silver/9465. 스티커/스티커.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int t, n;
    cin >> t;
    for (int i = 0; i < t; i++) {
        cin >> n;
        vector<vector<int>> arr(2, vector<int>(n, 0));
        vector<vector<int>> dp(3, vector<int>(n + 1, 0));
        for (int j = 0; j < 2; j++) {
            for (int k = 0; k < n; k++)
                cin >> arr[j][k];
        }
        for (int j = 1; j < n + 1; j++) {
            dp[0][j] = max(max(dp[0][j - 1], dp[2][j - 1]), max(dp[1][j - 1], dp[2][j - 1]));
            dp[1][j] = max(dp[0][j - 1], dp[2][j - 1]) + arr[0][j - 1];
            dp[2][j] = max(dp[0][j - 1], dp[1][j - 1]) + arr[1][j - 1];
        }
        cout << max(max(dp[0][n], dp[2][n]), max(dp[1][n], dp[2][n])) << "\n";
    }
    return 0;
}

File: 백준/Silver/9465. 스티커/스티커.py
import sys
input=lambda:sys.stdin.readline().rstrip()
t=int(input())
for _ in range(t):
    n=int(input())
    arr=[[*map(int,input().split())] for _ in range(2)]
    dp=[[0]*(n+1) for _ in range(3)]
    for i in range(1,n+1):
        dp[0][i]=max(dp[0][i-1],dp[1][i-1],dp[2][i-1])
        dp[1][i]=max(dp[0][i-1],dp[2][i-1])+arr[0][i-1]
        dp[2][i]=max(dp[0][i-1],dp[1][i-1])+arr[1][i-1]
    print(max(dp[0][n],dp[1][n],dp[2][n]))